### PR TITLE
Implement cloudstack datacenter config validator

### PR DESCRIPTION
**Pull request by wongni**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/2406

*Description of changes:*
Implemented CloudStack DataCenter Config's validator

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by maxdrib — APPROVED** [review]

File: pkg/providers/cloudstack/cloudstack.go [hunk 1]
```diff
@@ -453,6 +453,8 @@ func (p *cloudstackProvider) SetupAndValidateCreateCluster(ctx context.Context,
 		return fmt.Errorf("validating environment variables: %v", err)
 	}
 
+	p.datacenterConfig.SetDefaults()
```
Comment (jiayiwang7) [inline pkg/providers/cloudstack/cloudstack.go:456]: why are we adding this back?

File: pkg/providers/cloudstack/cloudstack.go [hunk 2]
```diff
@@ -492,6 +494,8 @@ func (p *cloudstackProvider) SetupAndValidateUpgradeCluster(ctx context.Context,
...
```
Comment (jiayiwang7) [inline pkg/providers/cloudstack/cloudstack.go:497]: same here

File: pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go [hunk 3]
```diff
@@ -160,10 +161,29 @@ func (v *CloudStackDatacenterConfig) Marshallable() Marshallable {
...
+	if len(v.Spec.Zones) > 0 {
+		return errors.New("zones must be empty")
+	}
+	if len(v.Spec.AvailabilityZones) == 0 {
+		return errors.New("availabilityZones must not be empty")
+	}
+	azSet := map[string]CloudStackAvailabilityZone{}
+	for _, az := range v.Spec.AvailabilityZones {
+		if _, exists := azSet[az.Name]; exists {
+			return errors.New(fmt.Sprintf("All availabilityZone names must be unique; found multiple %s", az.Name))
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go:182]:
```suggestion
			return fmt.Errorf("availabilityZone names must be unique. Duplicate name: %s", az.Name)
```

File: pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go [hunk 4]
```diff
@@ -160,10 +161,29 @@ func (v *CloudStackDatacenterConfig) Marshallable() Marshallable {
...
+	if v.Spec.ManagementApiEndpoint != "" {
+		return errors.New("managementApiEndpoint must be empty")
+	}
+	if len(v.Spec.Zones) > 0 {
+		return errors.New("zones must be empty")
+	}
+	if len(v.Spec.AvailabilityZones) == 0 {
+		return errors.New("availabilityZones must not be empty")
+	}
+	azSet := map[string]CloudStackAvailabilityZone{}
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go:179]: u can just define a "set" given not using the value. `azSet := map[string]bool`

**Review by jiayiwang7 — APPROVED** [review]

**Pull request merged**